index.js

```diff
@@ -40,7 +66,55 @@ app.post('/api/setMapping', function(req, res) {
     crossStreet: req.body.crossStreet,
     timestampCreated: req.body.timestampCreated,
     timestampSent: req.body.timestampSent,
+    source: req.body.source,
+    sourceID: req.body.sourceID,
   };
-  console.log(clientMapping);
-  res.send('MANYC Airtable-Gateway Client Mapping Set');
+  console.log(newRequest);
+  // TODO Create new request with Airtable API
+  res.send('MANYC Airtable-Gateway Needs Request Created');
 });
+
+/* ----------------------------------------------------------------*/
+/* Airtable Polling Section */
+/* ----------------------------------------------------------------*/
+// Config variables set via heroku
+const HEROKU_AIR_KEY = process.env.AIRTABLE_KEY;
+const HEROKU_AIR_BASE_ID = process.env.AIRTABLE_BASE;
+const HEROKU_TABLE_NAME = process.env.AIRTABLE_TABLE_NAME;
+
+const MANYC_NEW_REQ_URL = process.env.MANYC_NEW_REQ;
+const MANYC_UPDATE_REQ_URL = process.env.MANYC_UPDATE_REQ;
+const MANYC_DELETE_REQ_URL = process.env.MANYC_DELETE_REQ;
+
+const axios = require('axios');
+const Airtable = require('airtable');
+const base = new Airtable({apiKey: HEROKU_AIR_KEY}).base(HEROKU_AIR_BASE_ID);
+
+const ChangeDetector = require('airtable-change-detector');
+const airtableGatewayDetector = new ChangeDetector(
+  base(HEROKU_TABLE_NAME), {
+    writeDelayMs: 100,
+    metaFieldName: 'Meta', // Defaults to `Meta`
+    lastModifiedFieldName: 'Last Modified', // Defaults to `Last Modified`
+    lastProcessedFieldName: 'Last Processed',
+  },
+);
+
+airtableGatewayDetector.pollWithInterval(
+  'pollingNameForLogging',
+  10000, // interval in milliseconds
+  async (recordsChanged) => {
+    const numChanges = recordsChanged.length;
+    console.info(`Found ${numChanges} changes in ` + HEROKU_TABLE_NAME);
+    const promises = [];
+    recordsChanged.forEach((record) => {
+      console.log(record.fields);
+      promises.push(axios.post(MANYC_NEW_REQ_URL, {
+        // TODO Create Request Body
+      }));
+    });
+
+    // If doing many Airtable writes, be careful of 5rps rate limit
+    return Promise.all(promises).catch;
```

Comment:
Returning a Promise.all here doesn't seem right. Perhaps you meant to 'await' the Promise.all?
Also, why are you returning anything? This seems to be a timed-function that doesn't do anything with the return

Reply:
Yeah not sure, was doing this as instructed by the airtable-change-detector package. Not sure if it is expecting a return type / value. Can look into the airtable-change-detector package to see if we even need to return here

Reply:
Ok I took a look at the airtable-change-detector
as presumed, they are 'await'ing the promise that is returned'
[ChangeDetector.js line 104](https://github.com/crownheightsaid/airtable-change-detector/blob/6aa221337db1584cae792e8677fc63d9035448c8/src/ChangeDetector.js#L102)
the only thought I have left is that there shouldn't be a .catch and since there is no 'await'ing, it shouldn't be an async function.

Reply:
Gotcha removed .catch